src/modules/systemlib/battery.cpp

```diff
@@ -239,12 +230,8 @@ Battery::computeScale()
 	// reusing capacity calculation to get single cell voltage before drop
 	const float bat_v = _param_v_empty.get() + (voltage_range * _remaining_voltage);
 
-	_scale = _param_v_full.get() / bat_v;
-
-	if (_scale > 1.3f) { // Allow at most 30% compensation
-		_scale = 1.3f;
+	// Allow at most 30% compensation
+	// Shouldn't ever be more than the power at full battery
+	_scale = math::constrain(_param_v_full.get() / bat_v, 1.0f, 1.3f);
```

Comment:
You mean add the check to `constrain` (std::is_finite checks for inf & NaN)? Might makes sense. But that could also hide some problems.